### DISCUSSION TITLE
How to prevent Weblate (4.16.4) from transforming unsafe HTML chars with Kubernetes & Helm setup

### DISCUSSION THREAD

**Discussion by lendismart** [WeblateOrg/weblate#9333] [Q&A]
Hi 👋,

I'm having trouble figuring out why Weblate is transforming certain characters into their HTML safe encoding. For example the char  (') gets transformed into `&#39;`

Here's a screenshot to better illustrate this 👇

<img width="1716" alt="Screenshot 2023-05-31 at 13 29 57" src="https://github.com/WeblateOrg/weblate/assets/36134011/bb00c155-b037-469a-8b5d-13c0e6e6cefe">

What have I tried so far:

First of all, I would like to point out that I have installed Weblate using the Helm chart (https://artifacthub.io/packages/helm/weblate/weblate) in our own K8S cluster.

I assume this has something to do with the autofix features, more specifically with the `weblate.trans.autofixes.html.BleachHTML`. The thing is that if I try to disable it using the environment variables like this:

```
extraConfig:
  WEBLATE_REMOVE_AUTOFIX: weblate.trans.autofixes.html.BleachHTML
```

It does not work since in the 4.16.4 that class is not included in the list of classes that can be disabled as we can see [in this file](https://github.com/WeblateOrg/weblate/blob/weblate-4.16.4/weblate/settings_docker.py) on line 1082. 

```
# List of automatic fixups
AUTOFIX_LIST = [
    "weblate.trans.autofixes.whitespace.SameBookendingWhitespace",
    "weblate.trans.autofixes.chars.ReplaceTrailingDotsWithEllipsis",
    "weblate.trans.autofixes.chars.RemoveZeroSpace",
    "weblate.trans.autofixes.chars.RemoveControlChars",
]
modify_env_list(AUTOFIX_LIST, "AUTOFIX")
```

If I try to add it then I end up with an error when upgrading the Heml chart.

However, I did notice that in version 4.17 that class is indeed included in the `AUTOFIX_LIST`, but the latest chart version only supports 4.16.4

Here's the `AUTOFIX_LIST` in version 4.17:

```
# List of automatic fixups
AUTOFIX_LIST = [
    "weblate.trans.autofixes.whitespace.SameBookendingWhitespace",
    "weblate.trans.autofixes.chars.ReplaceTrailingDotsWithEllipsis",
    "weblate.trans.autofixes.chars.RemoveZeroSpace",
    "weblate.trans.autofixes.chars.RemoveControlChars",
    "weblate.trans.autofixes.chars.DevanagariDanda",
    "weblate.trans.autofixes.html.BleachHTML",
]
modify_env_list(AUTOFIX_LIST, "AUTOFIX")
```

Am I heading in the right direction here, or am I barking up the wrong tree? Any help will be greatly appreciated 🙏

**Comment by nijel** [comment]
The autofix is not involved here, as you didn't turn it on (see https://docs.weblate.org/en/latest/user/checks.html#unsafe-html-cleanup). Also, it would tell you that it was applied.

Are you getting this only from machine translation? I think we've addressed something like that recently, can you try upgrading?

**Reply by lendismart** [reply]
It only happens with machine translation.

How can I upgrade if I'm using the Helm chart? If I'm not mistaken the last published image is for 4.16.4-1 as you can see here: https://artifacthub.io/packages/helm/weblate/weblate.

**Reply by nijel** [reply]
It should be possible to change Docker tag you're using. Helm chart will be updated soon, but we're waiting for some fixes in the Docker container.
